### PR TITLE
Strong mode fixes and more

### PR DESCRIPTION
**Pull request by thosakwe**
1. Enable strong mode, add static type check fixes
2. Sort signature parameters lexicographically. Many OAuth providers expect this, ex. Upwork authentication doesn't work unless your keys in ABC order.
3. Fixed bugs in `test/` files
4. Remove check for contentLength when sending a body; the `http` package doesn't send one, which means that POST requests are always considered "improperly signed."
5. Sort parameters lexicographically in the server implementation as well.

All tests are passing, but it might also be a good idea to enable Travis CI or similar CI for this repo?

Thanks in advance.

### CODE REVIEW

File: lib/client.dart [hunk 1]
```diff
@@ -38,7 +38,7 @@ Map<String, String> generateParameters(
   
   if(request.contentLength != null
       && request.contentLength != 0
-      && ContentType.parse(request.headers["Content-Type"]).mimeType == "application/x-www-form-urlencoded") {
+      && ContentType.parse(request.headers["content-Type"]).mimeType == "application/x-www-form-urlencoded") {
```
Comment (erincandescent) [inline lib/client.dart]: I'm not sure why this is necessary, also "content-Type"

**Review by erincandescent — COMMENTED** [review]
Mostly LGTM but I have a question

**Review by erincandescent — COMMENTED** [review]
Wont sorting these change the signature value?

The spec doesn't say to do this to my memory

File: lib/client.dart (hunk 1 above)
Comment (thosakwe) [reply lib/client.dart]: Forgot why I did that, TBH

**Comment by thosakwe** [comment]
Great question. I don't remember, either; however, it seems like some OAuth providers sort them alphabetically as a means of canonicalizing how they accept the parameters. Other than this, I don't see a defined order in the OAuth spec, and the order in which this package sends keys might not be the order in which other servers expect it.

I'll test this out against other OAuth providers to see what happens. It might result in me implementing this functionality as a flag, i.e. `sortAlpha`

**Comment by erincandescent** [comment]
So I think the spec actually says to sort parameters: https://tools.ietf.org/html/rfc5849#section-3.4.1.3.2

But it also seems we already sort things: https://github.com/erincandescent/oauth.dart/blob/3e0211da298b6f563f09527c9bfa555989a098ad/lib/src/core.dart#L119

So I'm curious as to why we need to use a SplayTreeMap here. Do servers need the parameters in sorted order? Maybe other servers aren't sorting them as they're supposed to?

**Comment by thosakwe** [comment]
The issue is that with the current code, when query or body parameters are
added, those aren’t necessarily sorted.

A SplayTreeMap is the easiest way to ABC sort Maps in Dart, AFAIK.

TLDR; in the current state of this library, parameters are not always ABC
sorted

**Comment by thosakwe** [comment]
Ping @erincandescent

**Comment by thosakwe** [comment]
Ping

**Comment by thosakwe** [comment]
Ping

Anything we can do about this?

**Comment by thosakwe** [comment]
@erincandescent Hope everything is all right on your end.

**Comment by thosakwe** [comment]
Ping. Anything else you need me to do on my end?

**Comment by thosakwe** [comment]
For anybody else who was planning to use OAuth 1.0 in Dart, I’m hard forking this library, after 9 months of inactivity on this PR.
